src/d3a/models/market/__init__.py

```diff
@@ -60,6 +60,7 @@ def wrapper(self, *args, **kwargs):
 class Market:
 
     def __init__(self, time_slot=None, bc=None, notification_listener=None, readonly=False,
+                 grid_fee_type=ConstSettings.IAASettings.GRID_FEE_TYPE,
```

Comment:
Would be less error prone IMO if we discourage passing default anywhere other than `SimulationConfig`

Reply:
I assume you are talking about all market classes, right ? Reason that I added the default value was due to the order of the arguments. Even timeslot is a kwargs, therefore in order to put an argument after that, it has to provide a default value. The proper solution to that would be to remove default values for all parameters here, and reorder them (for instance, bc should not be the 2nd parameter IMO). Nevertheless, I will not do it in the context of this story since this change  is quite error prone. Better to do in its own ticket.

Reply:
Please create a ticket.

Reply:
https://gridsingularity.atlassian.net/browse/D3ASIM-2194